cli/src/funTest/kotlin/ExamplesFunTest.kt

```diff
@@ -159,6 +158,7 @@ class ExamplesFunTest : StringSpec() {
             )
 
             report shouldHaveSize 1
+            outputDir.deleteRecursively() shouldBe true
```

Comment:
The added assertion is not related to asserting what the test is about.
I'd prefer to not use assertions for that, but rather a `require` if really needed.
What do you think?

Have you considered `safelyDeleteRecursively(force=true)` instead?

Reply:
> The added assertion is not related to asserting what the test is about.

That depends a bit on how strict you want to be. To me, properly closing the file handle in `generateReport()` is something that makes sense to be tested in this context. And not closing the file handle properly would prevent deletion of the directory on Windows at least. So from that perspective, I'd argue it's fine to have this assertion here.

> I'd prefer to not use assertions for that, but rather a `require` if really needed.
> What do you think?

Besides my above argumentation, I'd prefer to not mix assertions with requirements in tests for clarity.

> Have you considered `safelyDeleteRecursively(force=true)` instead?

Good hint, done. Although it actually shouldn't be required here. But I still did that change for consistency with other places where we delete temporary directories in tests.

Reply:
> That depends a bit on how strict you want to be. 

I find it easier to grasp what the test assertions are if it suffices to look at all `shoulds` without the need for filtering that out.
I think it's best to ignore that topic in scope of this PR.